server/routes/createGoal/validators/GoalForm.test.ts

```diff
@@ -0,0 +1,112 @@
+import { validate } from 'class-validator'
+import { plainToInstance } from 'class-transformer'
+import { GoalCompleteDateOptions, GoalForm } from './GoalForm'
+import { flattenErrors } from '../../../middleware/validationMiddleware'
+
+describe('GoalForm', () => {
+  it('should validate given validators return no errors', async () => {
+    // Given
+    const body = {
+      title: 'Learn Spanish',
+      targetCompletionDateOption: GoalCompleteDateOptions.THREE_MONTHS,
+      steps: [{ title: 'Book Spanish course' }],
+    }
+
+    // When
+    const goalForm = plainToInstance(GoalForm, body)
```

Comment:
I think the CIAG UI project also used `plainToInstance` and it confused me then as it does now 😁 
I think (for me) my biggest bug-bear with it is that args appear to be the wrong way round 🤣 
The function name is saying "take a plain object and turn it into an instance of some type", so on that basis I'd expect it to read: `const goalForm = plainToInstance(body, GoalForm)` 😁 

Anyway, not to worry about that, I know it's a 3rd party library, and I'm sure more people are happy with it than just me moaning about it! 🤣 

My real question in this code is whether we need it at all (in this use case, not generally).
Could we not simply do:
```
    // Given
    const goalForm: GoalForm = {
      title: 'Learn Spanish',
      targetCompletionDateOption: GoalCompleteDateOptions.THREE_MONTHS,
      steps: [{ title: 'Book Spanish course' }],
    }

    // When
    const validationErrors = await validate(goalForm)
```

Would that work? 🤔

Reply:
It might work, but it would be validating against the untransformed body values.

`validationMiddleware.ts` will first transform the request body (which is a generic Record<string, string> type) to a GoalForm, then once transformed it runs class validator against the actual `GoalForm` object. Otherwise it would be validating against the request body which would just be strings.

It might work in this case because all the validation should work against strings, but in other examples we might be expecting date strings in the request body to first to be transformed into `Date` objects before running date validation against them.